pyogrio/raw.py

```diff
@@ -191,7 +206,93 @@ def read_arrow(
     dataset_kwargs = _preprocess_options_key_value(kwargs) if kwargs else {}
 
     try:
-        result = ogr_read_arrow(
+        with ogr_open_arrow(
+            path,
+            layer=layer,
+            encoding=encoding,
+            columns=columns,
+            read_geometry=read_geometry,
+            force_2d=force_2d,
+            skip_features=skip_features,
+            max_features=max_features or 0,
+            where=where,
+            bbox=bbox,
+            fids=fids,
+            sql=sql,
+            sql_dialect=sql_dialect,
+            return_fids=return_fids,
+        ) as source:
+            meta, reader = source
+            table = reader.read_all()
+    finally:
+        if buffer is not None:
+            remove_virtual_file(path)
+
+    return meta, table
+
+
+def open_arrow(
+    path_or_buffer,
+    /,
+    layer=None,
+    encoding=None,
+    columns=None,
+    read_geometry=True,
+    force_2d=False,
+    skip_features=0,
+    max_features=None,
+    where=None,
+    bbox=None,
+    fids=None,
+    sql=None,
+    sql_dialect=None,
+    return_fids=False,
```

Comment:
```suggestion
    return_fids=False,
    **kwargs,
```

Some fixes are needed to update for a recently merged PR: we need to accept kwargs here, and then convert that to `dataset_kwargs` variable

Reply:
I think I fixed this correctly